lib/convertkit/client.rb

```diff
@@ -24,12 +24,16 @@ def get_request(path)
       JSON.parse(json.body)
     end
 
-    def post_request(path, params)
+    def post_request(path, params, follow_redirects: false)
       url = "#{@uri}/v#{@version}#{path}?from=clickfunnels"
 
-      json = self.class.post(url, body: {
-        api_key: @key,
-        }.merge(params))
+      json = self.class.post(
+        url,
+        follow_redirects: follow_redirects,
```

Comment:
this is one of the option that `httparty` has as default `true` and in base on this option it request again in base on the redirection from the post request. In this case `convertkit` does not redirect so it loops 5 times into the same request and returns the error. Setting this to `false` should directly return the response without redirecting anywhere.